src/app/features/current-account/current-account-avatar.tsx

```diff
@@ -5,10 +5,14 @@ import { BoxProps } from '@stacks/ui';
 import { useCurrentAccountDisplayName } from '@app/common/hooks/account/use-account-names';
 import { useDrawers } from '@app/common/hooks/use-drawers';
 import { AccountAvatar } from '@app/components/account/account-avatar/account-avatar';
-import { useCurrentStacksAccount } from '@app/store/accounts/blockchain/stacks/stacks-account.hooks';
+import { useCurrentAccountIndex } from '@app/store/accounts/account';
+import { useStacksAccounts } from '@app/store/accounts/blockchain/stacks/stacks-account.hooks';
+import { StacksAccount } from '@app/store/accounts/blockchain/stacks/stacks-account.models';
 
 export const CurrentAccountAvatar = memo((props: BoxProps) => {
-  const currentAccount = useCurrentStacksAccount();
+  const accountIndex = useCurrentAccountIndex();
+  const accounts = useStacksAccounts();
+  const currentAccount = accounts[accountIndex] as StacksAccount | undefined;
```

Comment:
In future this could come from the rootkey's fingerprint, so that it doesn't target any specific chain.